packages/shared/src/element-reparenter.ts

```diff
@@ -0,0 +1,40 @@
+export default class ElementReparenter {
```

Comment:
This seems to be on the right track. Would be interested in how this API supports `overlay-trigger` or not as a confirmation of the approach.

Reply:
I think it should be pretty easy to rework the core reparenting logic here to use the same helper: https://github.com/adobe/spectrum-web-components/blob/main/packages/overlay/src/ActiveOverlay.ts#L328

I also know we were talking about potentially supporting an element array for the overlay contents, but that looks a lot more disruptive (e.g. it looks like there's some `delayed` attribute handling on the content element).